packages/core/src/shared/sam/build.ts

```diff
@@ -236,6 +236,7 @@ export async function runBuild(arg?: TreeNode): Promise<SamBuildResult> {
     } catch (error) {
         throw ToolkitError.chain(error, 'Failed to build SAM template', {
             details: { ...resolveBuildArgConflict(buildFlags) },
+            code: error instanceof ToolkitError ? error.code : undefined,
```

Comment:
ideally you would remove this logic into a function to do 
```
     code: getErrorCode(error)
```
instead of having to repeat the `instanceof` inline in 3 different places.

Reply:
Thanks. Updated